src/k8-metadata-client/src/client.rs

```diff
@@ -188,6 +188,14 @@ pub trait MetadataClient: Send + Sync {
                     );
                     let created_item = self.create_item(value).await?;
                     Ok(ApplyResult::Created(created_item))
+                } else if err.to_string() == "not found" {
```

Comment:
is there specific error that can be downcast? and what scenario

Reply:
This is for the cases where we return Err(anyhow!("not found")).

This happens only on DoNothingClient and MemoryClient (which don't rely on MetaStatus error.